src/Forms/ConfirmedPasswordField.php

```diff
@@ -562,9 +571,33 @@ public function saveInto(DataObjectInterface $record)
             return;
         }
 
-        if (!($this->canBeEmpty && !$this->value)) {
-            parent::saveInto($record);
+        // Create a random password if password is blank
+        if ($this->canBeEmpty && !$this->value) {
+            $this->value = $this->createStrongRandomPassword();
         }
+
+        parent::saveInto($record);
+    }
+
+    private function createStrongRandomPassword(): string
+    {
+        $value = '';
+        do {
+            // random_int() in OK to use for random password generation
+            $randomInt = random_int(0, PHP_INT_MAX);
+            // add in a-f as extra possible characters
+            $hashed = substr(md5($randomInt), 0, $this->getMaxLength() ?: 20);
+            // uppercase half of the characters
+            $hashed = implode('', [
+                strtoupper(substr($hashed, 0, floor(strlen($hashed) / 2))),
+                substr($hashed, ceil(strlen($hashed) / 2)),
+            ]);
+            // shuffle the characters
+            $arr = str_split($hashed);
+            shuffle($arr);
+            $value = implode('', $arr);
+        } while (!preg_match($this->strongPasswordRegex, $value));
```

Comment:
The regex btw is just "contains at least one letter and one number" (and ironically disallows special characters, so it's an awful regex for this purpose). So we just need to make sure there's at least one letter, at least one number, and no special characters (or update the regex to allow special chars by removing the `^` and `$` which just really ought not to be there)

Personally I think we should ignore that regex - it's only for the form field anyway, and we're not doing this for the sake of the form field, we're doing it for `Member`.

If we ignore the regex we can simply do this:

`$password = bin2hex(openssl_random_pseudo_bytes($length / 2));`

If we do want to respect that regex we can do this:

```php
// Start with a good base
$password = bin2hex(openssl_random_pseudo_bytes($maxLen / 4)); // this will end up being half the max length

// Ensure we have some letters to pass the regex
$letters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
$len = strlen($letters);
for ($i = 0; $i < floor($maxLen / 4); ++$i) {
    $password .= $letters[random_int(0, $len - 1)];
}

// Ensure we have some numbers to pass the regex
$numbers = '0123456789';
$len = strlen($numbers);
for ($i = 0; $i < floor($maxLen / 4); ++$i) {
    $password .= $numbers[random_int(0, $len - 1)];
}

// Make sure we don't reliably have <something> followed by letters followed by numbers
$password = str_shuffle($password); // though note that just like shuffle(), this isn't cryptographically secure
```

Reply:
I've redone the 'create password' method